planner/core/logical_plan_builder.go

```diff
@@ -299,7 +303,7 @@ func (p *LogicalJoin) extractOnCondition(conditions []expression.Expression, der
 					// false even if t.a is null or s.a is null. To make this join "empty aware",
 					// we should differentiate `t.a = s.a` from other column equal conditions, so
 					// we put it into OtherConditions instead of EqualConditions of join.
-					if binop.FuncName.L == ast.EQ && !arg0.InOperand && !arg1.InOperand {
+					if binop.FuncName.L == ast.EQ {
```

Comment:
Please move the comments here to the front as well.

Reply:
fixed